File: #069/Sqrt(x).cpp
class Solution {
public:
	int mySqrt(int x) {
		int a = x, b = x;
		while (a*a != x) {
			b = (a + x / b) / 2;
			b = b & 0x7FFFFFFF;
			if (b >= a) return a;
			a = b;
		}
		return a;
	}
};

File: #069/Sqrt(x).java
public class Solution {
    public int mySqrt(int num) {
        int x = num, y = num;
        while (x * x != num) {
            y = (x + num / x) / 2;
            y = y & 0x7FFFFFFF;
            if (y >= x) return x;
            x = y;
        }
        return x;
    }
}